### PR TITLE
refactor: migrate graphql-runner into TypeScript

### PR DESCRIPTION
**Pull request by rayriffy**
## Description

There's some changes in `IMatch` type because it originally says that context key data type can be only string `[key: string]: string` but context in line 55 of `graphql-runner.ts`data is in `boolean` type.

### CODE REVIEW

**Comment by LekoArts** [comment]
Hi, thanks for the PR!
As per https://github.com/gatsbyjs/gatsby/issues/21995 we want to use named exports, not default exports :)

**Comment by rayriffy** [comment]
This file is originally default export, Should I changed it to named export?

File: packages/gatsby/src/bootstrap/graphql-runner.ts [hunk 1]
```diff
@@ -60,3 +79,5 @@ module.exports = (store, reporter) => {
       return result
     })
 }
+
+export default graphQLRunner
```
Comment (blainekasten) [inline packages/gatsby/src/bootstrap/graphql-runner.ts:83]: we want to change these to named exports instead of default

**Comment by vladar** [comment]
Published in `gatsby@2.20.16`

**Pull request merged**